pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -419,6 +419,34 @@ func parseDnsNameToIpList(
 	config.DnsNameToIPList = nameToIPs
 }
 
+func parseStaticRoute(route *zconfig.IPRoute, config *types.NetworkInstanceConfig) error {
+	if route.DestinationNetwork == "" {
+		return errors.New("missing destination network address")
+	}
+	_, dstNetwork, err := net.ParseCIDR(route.DestinationNetwork)
+	if err != nil {
+		return fmt.Errorf("destination network is invalid: %w", err)
+	}
+	if route.Gateway == "" {
+		return errors.New("missing gateway IP address")
+	}
+	gatewayIP := net.ParseIP(route.Gateway)
+	if gatewayIP == nil {
+		return errors.New("gateway IP address is invalid")
+	}
+	if gatewayIP.IsUnspecified() {
+		return errors.New("gateway IP address is all-zeroes")
+	}
+	if gatewayIP.Equal(config.Gateway) {
+		return errors.New("static route gateway is the same as the network instance gateway")
```

Comment:
Why is this an error? Does the gateway always have to be external to the device and network instance? 
If so, shouldn't the check be that the gateway doesn't fall in the subnet assigned to the network instance?

Reply:
Gateway can be either external or one of the applications. So gateway inside the network instance subnet is OK.
However, I'm not sure if it would be wise to let the user create IP routes with the NI bridge as the gateway. Traffic routed by such route would arrive to EVE and then continue with the default route of the uplink, regardless of which uplink port is being used by NI at that moment. I'm not sure if we want to allow it. I though that it would be better to require that user enters specific external (or app) gateway.

Reply:
But they might not know the external gateway IP a priori whereas they can control the subnet assigned to a network instance. And most network instances have a single uplink.

Reply:
Valid point. I will make the necessary changes to allow this.

Reply:
Done